arcgis-ios-sdk-samples/Maps/Display device location with NMEA data sources/DisplayDeviceLocationWithNMEADataSourcesViewController.swift

```diff
@@ -24,31 +24,129 @@ class DisplayDeviceLocationWithNMEADataSourcesViewController: UIViewController {
     /// mock NMEA data every fixed amount of time.
     let mockNMEADataSource = SimulatedNMEADataSource(nmeaSourceFile: Bundle.main.url(forResource: "Redlands", withExtension: "nmea")!, speed: 1.5)
+    /// A formatter for the accuracy distance string.
+    let distanceFormatter: MeasurementFormatter = {
+        let formatter = MeasurementFormatter()
+        formatter.unitOptions = .naturalScale
+        formatter.numberFormatter.minimumFractionDigits = 1
+        formatter.numberFormatter.maximumFractionDigits = 1
+        return formatter
+    }()
     
     // MARK: Actions
     
-    @IBAction func start() {
+    /// Get the first connected and supported Bluetooth accessory with its
+    /// protocol string.
+    /// - Returns: A tuple of the accessory and its protocol,
+    ///            or nil if no supported accessory exists.
+    func firstSupportedAccessoryWithProtocol() -> (EAAccessory, String)? {
+        for accessory in EAAccessoryManager.shared().connectedAccessories {
+            // The protocol string to establish the EASession.
+            guard let protocolString = accessory.protocolStrings.first(where: { supportedProtocolStrings.contains($0) }) else {
+                // Skip any device which protocol is not included in the plist.
+                // This typically shouldn't happen, unless the device requires
+                // additional configuration.
+                continue
+            }
+            // Only return the first connected and supported accessory.
+            return (accessory, protocolString)
+        }
+        return nil
+    }
+    
+    /// The Bluetooth accessory picker connected to a supported accessory.
+    func accessoryDidConnect(connectedAccessory: EAAccessory, protocolString: String) {
+        nmeaLocationDataSource = AGSNMEALocationDataSource(eaAccessory: connectedAccessory, protocol: protocolString)
+        nmeaLocationDataSource?.locationChangeHandlerDelegate = self
+        start()
+    }
+    
+    @IBAction func chooseDataSource(_ sender: UIBarButtonItem) {
+        let alertController = UIAlertController(
+            title: "Choose an NMEA data source.",
+            message: nil,
+            preferredStyle: .actionSheet
+        )
+        // Add real data source to the options.
+        let realDataSourceAction = UIAlertAction(title: "Device", style: .default) { [self] _ in
+            if let (accessory, protocolString) = firstSupportedAccessoryWithProtocol() {
+                // Use the supported accessory directly if it's already connected.
+                accessoryDidConnect(connectedAccessory: accessory, protocolString: protocolString)
+            } else {
+                // Display an alert to pair the device with a Bluetooth accessory.
+                EAAccessoryManager.shared().showBluetoothAccessoryPicker(withNameFilter: nil) { error in
+                    if let error = error as? EABluetoothAccessoryPickerError {
```

Comment:
> ...the only kind of error returned is EABluetoothAccessoryPickerError

Yes, per the [`EABluetoothAccessoryPickerError`](https://developer.apple.com/documentation/externalaccessory/eabluetoothaccessorypickererror) doc.

> If the error code is alreadyConnected, we should just be able to use the connected accessory, correct?

Yes. But instead of trying to init the NMEA LDS after an `alreadyConnected` error code is returned, I attempted to init the LDS from potential connected accessory before showing the picker. Not sure if it is the best practice...